False positive in rule "vue/require-prop-types"

<!--
    This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

    1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 4.14.0
* **eslint-plugin-vue Version:** 4.0.0
* **Node Version:** 8.5.0

**Please show your full configuration:**
```js
// basic configuration
module.exports = {
  root: true,
  extends: [
    'airbnb-base',
    'plugin:vue/recommended',
  ],
  parserOptions: {
    parser: 'babel-eslint'
  },
  plugins: ['vue']
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
import props from './Foo.props';

// @vue/component
export default {
  name: 'Foo',
  props,
  render: h => h('div', 'foo'),
};
```

**What did you expect to happen?**
Should not raise require-prop-types error

**What actually happened? Please include the actual, raw output from ESLint.**
Setting props in a component that are defined outside of it raises the error:

```
C:\dev\foobar\src\app.js
  6:3   error  Props should at least define their types vue/require-prop-types

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/rules/require-prop-types.js
+++ b/lib/rules/require-prop-types.js
@@ -64,7 +64,9 @@ function create (context) {
 
     if (node.value.type === 'ObjectExpression') {
       checkProperties(node.value.properties)
-    } else {
+    }
+
+    if (node.value.type === 'ArrayExpression') {
       context.report({
         node,
         message: 'Props should at least define their types.'

--- a/tests/lib/rules/require-prop-types.js
+++ b/tests/lib/rules/require-prop-types.js
@@ -69,6 +69,24 @@ ruleTester.run('require-prop-types', rule, {
         }
       `,
       parserOptions: { ecmaVersion: 6, sourceType: 'module' }
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          props
+        }
+      `,
+      parserOptions: { ecmaVersion: 6, sourceType: 'module' }
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          props: externalProps
+        }
+      `,
+      parserOptions: { ecmaVersion: 6, sourceType: 'module' }
     }
   ],
 

